### ISSUE TITLE
New data sources

### ISSUE THREAD

**Issue by dirkkelly** [BetaNYC/citygram-nyc#41]
@bjelline can you please provide in this issue the configuration options required for implementing your work over at https://github.com/bjelline/nyc-311-selected

**Comment by bjelline** [comment]
I guess there are two steps:

1) copying the logos into app/assets/img/icons/   
2) inserting a "publisher" directly into postgres for each feed:

    insert into publishers (title,endpoint,active,icon,city) 
     values ('Rodents', 'https://nyc-311-selected.herokuapp.com/nyc-311-rodent', True, 'rat.png', 'NYC');


of yourse you could also fork nyc-311-selected and run it on your own heroku app.

**Comment by fma2** [comment]
@dirkkelly @bjelline Does it make sense to and/or would it be possible to add this work to https://github.com/citygram/citygram-services?

https://citygram-services.herokuapp.com/

**Comment by dirkkelly** [comment]
@fma2 I we're already using data from this source.

**Comment by bjelline** [comment]
@fma2 yes, I think it would make sense.  citygram-services adds caching and avoids some code duplication that is pressent in my implementation of the feed.  I'll have a look at it

**Comment by dirkkelly** [comment]
Some development notes from getting this publisher into my **development** instance.

We need to create a new publisher

```ruby
Citygram::DatabaseHelper.database[:publishers].insert(
  visible: true,
  active: true,
  title: 'Rodents',
  endpoint: 'https://nyc-311-selected.herokuapp.com/nyc-311-rodent',
  city: 'new-york',
  icon: 'rat.png'
)
```

that's useful code, let's add it to a seed so we can run it later

```
desc 'Seed the data'
task :seed do |t, args|
...
end
```

Then we run the seed

```
be rake db:seed
```

Loading the page will explode due to no image, download [rat.png](https://github.com/bjelline/nyc-311-selected/blob/master/public/rat.png)

![rat.png](https://raw.githubusercontent.com/bjelline/nyc-311-selected/master/public/rat.png)

No events in the system, need to update publishers

```
be rake publishers:update
```

That did it!

![screen shot 2015-02-23 at 10 56 05 pm](https://cloud.githubusercontent.com/assets/81055/6342998/3a3fad32-bbaf-11e4-9d70-42ec3d2684d7.png)

Next up! I can now run this on staging

**Comment by dirkkelly** [comment]
```bash
heroku fork -a citygram-nyc-staging citygram-nyc-311
heroku run rake console -a citygram-nyc-311
```

```ruby
require_relative 'lib/database_helper'

Citygram::DatabaseHelper.database[:publishers].insert(
  visible: true,
  active: true,
  title: 'Rodents',
  endpoint: 'https://nyc-311-selected.herokuapp.com/nyc-311-rodent',
  city: 'new-york',
  icon: 'rat.png'
)

Citygram::DatabaseHelper.database[:publishers].insert(
  visible: true,
  active: true,
  title: 'Water',
  endpoint: 'https://nyc-311-selected.herokuapp.com/nyc-311-water',
  city: 'new-york',
  icon: 'water.png'
)

Citygram::DatabaseHelper.database[:publishers].insert(
  visible: true,
  active: true,
  title: 'Air',
  endpoint: 'https://nyc-311-selected.herokuapp.com/nyc-311-air',
  city: 'new-york',
  icon: 'air.png'
)

Citygram::DatabaseHelper.database[:publishers].insert(
  visible: true,
  active: true,
  title: 'Pothole',
  endpoint: 'https://nyc-311-selected.herokuapp.com/nyc-311-street',
  city: 'new-york',
  icon: 'pothole.png'
)
```

```bash
heroku run rake publishers:update -a citygram-nyc-311
```

Had to turn on workers 

```bash
heroku ps:scale worker=1 -a citygram-nyc-311
# ... drink some tea
```

https://citygram-nyc-311.herokuapp.com/

**Comment by dirkkelly** [comment]
@bjelline There's a preview of the feeds available up on heroku, fantastic work!

https://citygram-nyc-311.herokuapp.com/

**Comment by bjelline** [comment]
Thanks!  Now we have to make sure people understand that those four feeds are sub-feeds of 311. How about this wording:

![screen shot 2015-02-24 at 04 58 48](https://cloud.githubusercontent.com/assets/36773/6346995/4a175880-bbe2-11e4-8a46-b46a1a678f68.png)

**Comment by bjelline** [comment]
I've ported two of the feeds to citigram-services:

* https://nyc-311-selected-services.herokuapp.com/nyc-311-water
* https://nyc-311-selected-services.herokuapp.com/nyc-311-rodents

Source here https://github.com/bjelline/citygram-services
